client/components/Editor/index.jsx

```diff
@@ -20,42 +21,100 @@ class Editor extends Component {
     constructor(props) {
         super(props);
 
-        const scenarioId = this.props.match.params.id || 'new';
-
         this.copyScenario = this.copyScenario.bind(this);
         this.deleteScenario = this.deleteScenario.bind(this);
-        this.updateScenario = this.updateScenario.bind(this);
-        this.onClick = this.onClick.bind(this);
-        this.onClickScenarioAction = this.onClickScenarioAction.bind(this);
-        this.getSubmitCallback = this.getSubmitCallback.bind(this);
+        this.getAllTabs = this.getAllTabs.bind(this);
         this.getPostSubmitCallback = this.getPostSubmitCallback.bind(this);
+        this.getSubmitCallback = this.getSubmitCallback.bind(this);
         this.getTab = this.getTab.bind(this);
-        this.getAllTabs = this.getAllTabs.bind(this);
+        this.onClick = this.onClick.bind(this);
+        this.onClickScenarioAction = this.onClickScenarioAction.bind(this);
+        this.setActiveView = this.setActiveView.bind(this);
         this.updateEditorMessage = this.updateEditorMessage.bind(this);
+        this.updateScenario = this.updateScenario.bind(this);
+
+        let { activeTab, activeSlideIndex, match, scenarioId } = this.props;
```

Comment:
I believe we can get `isNewScenario` here for new scenarios, which can help make some of the logic below a little since we don't need to check the match params.

Reply:
ie we pass `isNewScenario = true` in the props for the "Create A Moment" route

Reply:
Did it!